smack-core/src/main/java/org/jivesoftware/smack/altConnections/HttpLookupMethod.java

```
@@ -0,0 +1,174 @@
+     * @return list of endpoints
+     * @throws IOException exception due to input/output operations
+     * @throws XmlPullParserException exception encountered during XML parsing
+     * @throws URISyntaxException exception to indicate that a string could not be parsed as a URI reference
+     */
+    public static List<URI> lookup(DomainBareJid xmppServiceAddress, String relation) throws IOException, XmlPullParserException, URISyntaxException {
+        try (InputStream inputStream = getXrdStream(xmppServiceAddress)) {
+            XmlPullParser xmlPullParser = PacketParserUtils.getParserFor(inputStream);
+            List<URI> endpoints = parseXrdLinkReferencesFor(xmlPullParser, relation);
+            return endpoints;
+        }
+    }
+
+    /**
+     * Get remote endpoints for the given LinkRelation from host.
+     *
+     * @param xmppServiceAddress address of host
+     * @param relation {@link LinkRelation} specifying type of connection
+     * @return list of endpoints
+     * @throws IOException exception due to input/output operations
+     * @throws XmlPullParserException exception encountered during XML parsing
+     * @throws URISyntaxException exception to indicate that a string could not be parsed as a URI reference
+     */
+    public static List<URI> lookup(DomainBareJid xmppServiceAddress, LinkRelation relation) throws IOException, XmlPullParserException, URISyntaxException {
+        return lookup(xmppServiceAddress, relation.attribute);
+    }
+
+    /**
+     * Constructs a HTTP connection with the host specified by the DomainBareJid
+     * and retrieves XRD element in the form of an InputStream.
+     *
+     * @param xmppServiceAddress address of host
+     * @return InputStream containing XRD element
+     * @throws IOException exception due to input/output operations
+     */
+    public static InputStream getXrdStream(DomainBareJid xmppServiceAddress) throws IOException {
```

How does this method behave if no XRD document is published? This should be documented. I think it would ideally return `null` in this case.

---

Forget that about 'null', simply let the method throw.